ui/app/adapters/job.js

```diff
@@ -159,4 +167,138 @@ export default class JobAdapter extends WatchableNamespaceIDs {
       },
     });
   }
+
+  runAction(job, action, allocID) {
+    let messageBuffer = '';
+
+    const protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
+    const region = this.system.activeRegion;
+    const applicationAdapter = getOwner(this).lookup('adapter:application');
+    const prefix = `${
+      applicationAdapter.host || window.location.host
+    }/${applicationAdapter.urlPrefix()}`;
+
+    const wsUrl =
+      `${protocol}//${prefix}/job/${encodeURIComponent(job.get('id'))}/action` +
+      `?namespace=${job.get('namespace.id')}&action=${
+        action.name
+      }&allocID=${allocID}&task=${action.task.name}&group=${
+        action.task.taskGroup.name
+      }&tty=true&ws_handshake=true` +
+      (region ? `&region=${region}` : '');
+
+    // const socket = new WebSocket(wsUrl);
+    let socket;
```

Comment:
Does it make sense to eventually try to use the sockets service here?

Reply:
Ended up starting/stopping on this a couple times today; I think it's a good idea and will be marking it a TODO but I'm worried about drift from the exec implementation enough that I want to give it more consideration.